goal_src/engine/target/logic-target.gc

```diff
@@ -6,38 +6,3086 @@
+         (s0-6 (-> arg0 control transv y))
+         )
+    (set! sv-160 (-> arg0 control transv z))
+    (set!
+     sv-176
+     (vector-dot
+      (-> arg0 control dynam gravity-normal)
+      (-> arg0 control transv)
+      )
+     )
+    (let ((a0-36 (new-stack-vector0)))
+     (let
+      ((f0-17
+        (vector-dot
+         (-> arg0 control dynam gravity-normal)
+         (-> arg0 control transv)
+         )
+        )
+       )
+      0.0
+      (vector-!
+       a0-36
+       (-> arg0 control transv)
+       (vector-float*! a0-36 (-> arg0 control dynam gravity-normal) f0-17)
+       )
+      )
+     (let ((f0-18 (vector-length a0-36)))
+      f0-18
+      (let ((t2-4 f0-18))
+       (s4-6 s3-6 s2-6 s1-6 s0-6 sv-160 sv-176 t2-4)
+       )
+      )
+     )
+    )
+   (format
+    arg1
+    "~0ky:~,,2M t:~d cy: ~,,2M my: ~,,2M impv:~,,2M "
+    (-> arg0 control unknown-vector52 y)
+    (-
+     (-> *display* base-frame-counter)
+     (the-as uint (-> arg0 control unknown-dword10))
+     )
+    (- (-> arg0 control trans y) (-> arg0 control unknown-vector52 y))
+    (-
+     (-> arg0 control unknown-vector111 y)
+     (-> arg0 control unknown-vector110 y)
+     )
+    (-> arg0 control ground-impact-vel)
+    )
+   (format
+    arg1
+    "~0k~Tcd: ~,,2M md:~,,2M~%"
+    (vector-xz-length
+     (vector-!
+      (new-stack-vector0)
+      (-> arg0 control trans)
+      (-> arg0 control unknown-vector52)
+      )
+     )
+    (vector-xz-length
+     (vector-!
+      (new-stack-vector0)
+      (-> arg0 control unknown-vector111)
+      (-> arg0 control unknown-vector110)
+      )
+     )
+    )
+   (format
+    arg1
+    "~0kbend b:~F t:~F s:~F~%"
+    (-> arg0 control unknown-float80)
+    (-> arg0 control unknown-float81)
+    (-> arg0 control unknown-float82)
+    )
+   (dummy-10 (-> arg0 skel) arg1)
+   )
+  (draw-history (-> arg0 control))
+  #f
   )
 
-(defun start ((game-mode symbol) (cont continue-point))
-  "Start the target system"
+;; definition for function read-pad
+;; Used lq/sq
+(defbehavior read-pad target ((arg0 vector))
+  (when (!= (-> self control unknown-dword00) (-> *display* real-frame-counter))
+   (set!
+    (-> self control unknown-vector31 quad)
+    (-> self control unknown-vector30 quad)
+    )
+   (set! (-> self control unknown-float21) (-> self control unknown-float20))
+   (set! (-> self control unknown-dword00) (-> *display* real-frame-counter))
+   )
+  (set! (-> arg0 x) (sin (-> self control unknown-cpad-info00 stick0-dir)))
+  (set! (-> arg0 y) 0.0)
+  (set! (-> arg0 z) (cos (-> self control unknown-cpad-info00 stick0-dir)))
+  (set! (-> arg0 w) 0.0)
+  (set! (-> self control unknown-vector30 quad) (-> arg0 quad))
+  (set!
+   (-> self control unknown-float20)
+   (-> self control unknown-cpad-info00 stick0-speed)
+   )
+  (let ((s5-0 vector-matrix*!)
+        (s4-0 arg0)
+        (t9-2 matrix-local->world)
+        (a0-11 #t)
+        )
+   (s5-0 s4-0 arg0 (t9-2 a0-11))
```

Comment:
This looks weird to me. I looked at the assembly and I think that `matrix-local->world` takes a second `symbol` argument that is unused. If you add that, hopefully this `let` will be eliminated.